src/js/index.js

```diff
@@ -0,0 +1,102 @@
+import { XApp } from "./xapp/index.js";
+import { ServerMenuAPI } from "./api.js";
+
+function createApp() {
+    const api = new ServerMenuAPI();
+
+    const app = new XApp({
+        el: document.getElementById("app"),
+    });
+
+    app.setData({
+        categories: [],
+        categorySelected: {
+            name: "",
+            key: "",
+        },
+        menuList: [],
+        menuInput: "",
+        menuCount: 0,
+    });
+
+    app.setMethods({
+        async createMenu(e) {
+            e.preventDefault();
+            if (this.data.menuInput !== "") {
+                try {
+                    const newMenu = await api.createMenu(this.data.categorySelected.key, this.data.menuInput);
+                    this.data.menuList.push(newMenu);
+                    this.data.menuCount++;
+                    this.data.menuInput = "";
+                } catch (e) {
+                    alert(e.data.message);
+                    return;
+                }
+
+                this.render();
+                this.elements.menuInput.focus();
+            }
+        },
+        onMenuInputChange(e) {
+            this.data.menuInput = e.target.value;
+        },
+        async editMenu(e, item) {
+            const changed = prompt("새로운 메뉴 이름을 입력해주세요.", item.content);
+            if (changed) {
+                try {
+                    await api.editMenu(this.data.categorySelected.key, { ...item, content: changed });
```

Comment:
이건 제 개인 취향이지만.. 객체로 넘기는 값이 길 경우 줄바꿈을 해주면 조금 더 읽기 깔끔할 것 같아요 👍 프리티어 printWidth 설정상 자동 포맷팅이 돼서 그러는 것 같긴 한데 리뷰할땐 조금 잘려보일 수 있어 작성해봅니다
```suggestion
 await api.editMenu(this.data.categorySelected.key, {
      ...item,
      content: changed,
 });
```